clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11630,6 +11630,10 @@ def err_sycl_expected_finalize_method : Error<
 def ext_sycl_2020_attr_spelling : ExtWarn<
   "use of attribute %0 is a SYCL 2020 extension">,
   InGroup<Sycl2017Compat>;
+def err_sycl_taking_address_of_wrong_function : Error<
+  "taking address of a function not marked with "
+  "`intel::device_indirectly_callable attribute` is not allowed in SYCL device "
```

Comment:
is there a reason to pass the attribute directly instead of %0 attribute here?

Reply:
Actually I didn't even think about that, but I'm not quite sure it is necessary, because at the point of diagnostic emission we don't have an object of attribute, so we will need to construct it and there is no other cases where some other missing attribute would cause the error.

Reply:
It is not necessary.  

>>because at the point of diagnostic emission we don't have an object of attribute, so we will need to construct it and there is no other cases where some other missing attribute would cause the error.

Agreed